File: server/src/types/steam.py
from pydantic import BaseModel
from typing import Optional
    
class SteamPlayer(BaseModel):
    steamid: str
    persona_name: str
    profile_url: str
    avatar_full: str
    real_name: str
    country_code: str
    state_code: str

class Category(BaseModel):
    id: int
    description: str

class Genre(BaseModel):
    id: str
    description: str

class PriceOverview(BaseModel):
    currency: str
    price_in_cents: int
    final_formatted: str
    discount_percentage: int

class Metacritic(BaseModel):
    score: int
    url: str

class SteamGame(BaseModel):
    appid: int
    game_type: str
    game_name: str
    is_free: Optional[bool]
    detailed_description: str
    header_image: str
    website: Optional[str] = None
    recommendations: Optional[int] = None
    release_date: Optional[str] = None
    esrb_rating: Optional[str] = None
    developers: list[str]
    publishers: list[str]
    categories: list[Category]
    genres: list[Genre]
    price_overview: Optional[PriceOverview] = None
    metacritic: Optional[Metacritic] = None
    screenshots: list[str]
    tags: list[str]

class Wishlist(BaseModel):
  steamid: int
  appid: int
  priority: int
  
class DealsGGPrices(BaseModel):
  retail_price: float
  retail_price_low: float
  keyshop_price: float
  keyshop_price_low: float
  
class DealsGG(BaseModel):
  appid: int
  name: str
  url: str
  image_url: str
  prices: DealsGGPrices
  currency: str
  
class AppIdsRequest(BaseModel):
    appids: list[int]

def correct_user_account_response() -> dict:
    return {
        "response": {
            "players": []
        }
    }
    
def correct_user_wishlist_response() -> dict:
    return {
        "response": {
            "items": []
        }
    }
    
def correct_game_data_response(appid: str) -> dict:
    return {
        appid: {
            "data": {}
        }
    }

File: client/src/types/steam.ts
export interface SteamPlayer {
  steamid: string;
  communityvisibilitystate: number;
  profilestate: number;
  personaname: string;
  profileurl: string;
  avatar: string;
  avatarmedium: string;
  avatarfull: string;
  avatarhash: string;
  lastlogoff: number;
  personastate: number;
  realname?: string;
  primaryclanid?: string;
  timecreated?: number;
  personastateflags?: number;
  loccountrycode?: string;
  locstatecode?: string;
}

export interface Category {
  id: number;
  description: string;
}

export interface Genre {
  id: string;
  description: string;
}

export interface PriceOverview {
  currency: string;
  price_in_cents: number;
  final_formatted: string;
  discount_percentage: number;
}

export interface Metacritic {
  score: number;
  url: string;
}

export interface SteamGame {
  appid: number;
  game_type: string;
  game_name: string;
  is_free: boolean;
  detailed_description: string;
  header_image: string;
  website?: string;
  recommendations: number;
  release_date: string;
  esrb_rating: string;
  developers: string[];
  publishers: string[];
  categories: Category[];
  genres: Genre[];
  price_overview: PriceOverview;
  metacritic?: Metacritic;
  screenshots: string[];
  tags: string[];
}

export interface WishlistItem {
  steamid: number
  appid: number
  priority: number
}

export interface DealsGGPrices{
  retail_price: number;
  retail_price_low: number;
  keyshop_price: number;
  keyshop_price_low: number;
}
  
export interface DealsGGItem{
  appid: number;
  name: string;
  url: string;
  image_url: string;
  prices: DealsGGPrices;
  currency: string;
}